### ISSUE TITLE
Build fail ACE (6.4.0)  MacOS 10.12 (16A270f)

### ISSUE THREAD

**Issue by Selenium** [Homebrew/homebrew-core#3639]
### Bug reports:

Hello, 
Since the first beta os macOS Sierra the formula ACE 6.4.0 dosent build anymore.

`brew gist-logs <formula>`
https://gist.github.com/anonymous/9ec22f126918e6f93567d6ff9ebca1eb

**Comment by mistydemeo** [comment]
We need to backport this commit: https://github.com/DOCGroup/ACE_TAO/commit/a2b3e76ab4407981816cfa903b9fc749b8c53886

However, even with that, there's a bug relating to a clock type: https://gist.github.com/0caa29006625fb87d6ff62c16096389a

```
clang++  -m64 -W -Wall -DACE_HAS_CUSTOM_EXPORT_MACROS=0  -DNDEBUG  -I/private/tmp/ace-20160810-
7106-hswow7/ACE_wrappers -DACE_NDEBUG -D__ACE_INLINE__ -I.. -DACE_BUILD_DLL  -c  -o .shobj/Name
_Proxy.o Name_Proxy.cpp
In file included from Name_Proxy.cpp:1:
In file included from /private/tmp/ace-20160810-7106-hswow7/ACE_wrappers/ace/Name_Proxy.h:21:
In file included from /private/tmp/ace-20160810-7106-hswow7/ACE_wrappers/ace/INET_Addr.h:15:
In file included from /private/tmp/ace-20160810-7106-hswow7/ACE_wrappers/ace/Sock_Connect.h:23:
In file included from /private/tmp/ace-20160810-7106-hswow7/ACE_wrappers/ace/Basic_Types.h:52:
In file included from /private/tmp/ace-20160810-7106-hswow7/ACE_wrappers/ace/os_include/os_limi
ts.h:25:
In file included from /private/tmp/ace-20160810-7106-hswow7/ACE_wrappers/ace/os_include/os_unis
td.h:25:
/private/tmp/ace-20160810-7106-hswow7/ACE_wrappers/ace/os_include/sys/os_types.h:49:16: error:
typedef redefinition with different types ('int' vs 'enum clockid_t')
   typedef int clockid_t;
               ^
/usr/include/time.h:171:3: note: previous definition is here
} clockid_t;
  ^
```

**Comment by claudiofantacci** [comment]
Hello everyone, are there any news about this issue? :-)

**Comment by jwillemsen** [comment]
A new ACE 6.4.1 with the necessary changes is scheduled for in about 1 to 2 weeks of time

**Comment by claudiofantacci** [comment]
Cool! Thanks @jwillemsen !

**Comment by jwillemsen** [comment]
New ACE version is available, created pull request  #5752

**Comment by claudiofantacci** [comment]
Thanks @jwillemsen!

**Comment by dunn** [comment]
Fixed by  #5752.
